application/models/QuestionGroup.php

```diff
@@ -163,14 +163,16 @@ public function insertRecords($data)
      * Deletes a question group and all its dependencies.
      * Returns affected rows of question group table (should be 1 or null)
      * @param integer $groupId
-     * @param integer $surveyId
+     * @param integer|null $surveyId deprecated
      * @return int|null
      */
-    public static function deleteWithDependency($groupId, $surveyId)
+    public static function deleteWithDependency($groupId, $surveyId = null)
     {
+        $QuestionGroup ::self::model()->findByPk($groupId);
         // Abort if the survey is active
-        $surveyIsActive = Survey::model()->findByPk($surveyId)->active !== 'N';
+        $surveyIsActive = Survey::model()->findByPk($QuestionGroup->sid)->active !== 'N';
         if ($surveyIsActive) {
+            /* TODO : replace by a cleaner way to send error : model muts not use HTML only error system */
             Yii::app()->user->setFlash('error', gT("Can't delete question group when the survey is active"));
```

Comment:
@gabrieljenik do you think it's better to send a 500 error here ? Check of active survey must be done by controller in my opinion. Then if it(s not done : it' a 500

Reply:
It's done in LimeSurvey\Models\Services\QuestionGroupService->checkDeletePermission i throw a 500 here